general_stats_generator.py

```diff
@@ -0,0 +1,33 @@
+from functools import reduce
+from services import DataServices
+
+class GeneralStatsCreator(DataServices):
```

Comment:
I think it'd be good to define some kind of abstract class for that, as you have 3 stats generators with some kind of common "generate" method